### PR TITLE
Persist sorting in Browse models

### LINKED ISSUE #42952: [Browse] Preserve sort on page reload

### PR DESCRIPTION
**Pull request by rafpaf**
This branch persists sorting of Browse models table in the API and adds an e2e test

Closes #42952

### CODE REVIEW

**Comment by rafpaf** [comment]
* **#42537** <a href="https://app.graphite.dev/github/pr/metabase/metabase/42537?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rafpaf and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/metabase/models/user.clj [hunk 1]
```diff
@@ -518,6 +518,22 @@
...
+  :default    "collection")
+
+(defsetting browse-models-sort-direction
+  (deferred-tru "User preference for the sort direction for the 'Browse models' table")
+  :user-local :only
+  :export?    false
+  :visibility :authenticated
+  :type       :string
+  :default    "asc")
+
```
Comment (rafpaf) [inline src/metabase/models/user.clj:536]: These are so closely related I thought about storing them together as a JSONified string, or as a string like "+name" or "-name", but this flat, explicit approach seems simplest

Comment (dpsutton) [reply src/metabase/models/user.clj:536]: We often will used a closed set of valid choices here. No worries not to, but need to make sure that the FE can handle invalid input.

Comment (rafpaf) [reply src/metabase/models/user.clj:536]: If the sort column is invalid - neither "asc" or "desc" - then the FE defaults to ascending

Comment (dpsutton) [reply src/metabase/models/user.clj:536]: and also the column to sort on? (no need to respond, just making sure we are defensive)

Comment (rafpaf) [reply src/metabase/models/user.clj:536]: Ah yes, to correct my earlier statement: if the sort _direction_ in the API is invalid (or absent) we default to `asc`.

If the sort _column_ in the API is invalid (or absent), we default to `collection`.

**Review by iethree — CHANGES_REQUESTED** [review]
1) I think persisting sort direction is not a priority for this feature, and introduces unnecessary complexity, for a tiny benefit to users that they will not likely notice.
2) this likely adds complexity to improving performance, so I'd prefer to defer this until that is sorted

**Comment by rafpaf** [comment]
Pausing work on this because it runs into a race condition issue (which also affects the Browse filter toggle) and we need to iron that out first

**Comment by rafpaf** [comment]
Closing for now. Will return to this if the priority is determined to be higher.

**Pull request closed without merging**